os/android/iahwc2.cpp

```diff
@@ -230,19 +230,34 @@ static inline void supported(char const *func) {
 HWC2::Error IAHWC2::CreateVirtualDisplay(uint32_t width, uint32_t height,
                                          int32_t *format,
                                          hwc2_display_t *display) {
-  *display = (hwc2_display_t)(virtual_display_index_ + HWC_DISPLAY_VIRTUAL +
-                              VDS_OFFSET);
+  spin_lock_.lock();
```

Comment:
Identify the shared resources utilized within CreateVirtualDisplay(), and limit the lock for the duration of when shared resources are being accessed, unless it is necessary to secure the lock in the entire span of the function.

Reply:
Thanks reminds on this. Just shrank the critical region little bits in updated PR.